doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -1462,7 +1522,21 @@ inhabitant of the type) is interactively built using tactics. The interactive
 proof mode is described in Chapter :ref:`proofhandling` and the tactics in
 Chapter :ref:`Tactics`. The basic assertion command is:
 
-.. cmd:: Theorem @ident {? @binders } : @type
+.. cmd:: @thm_token @ident_decl {* @binder } : @term {* with @ident_decl {* @binder } : @term }
```

Comment:
This is a case where the use of the `with` clause changes the semantics so significantly that we should really document the two variants separately (as separate commands, with separate indexes).

I've tried in the past to deprecate `Theorem ... with` without success (see #8265). More generally, I've tried to make the syntax of `Definition` and `Theorem` closer (see #6567 and the plan to support the `Theorem ... := ...` syntax), but did not make much progress so far...

Reply:
This applies not just `Theorem ... with` but to `Lemma, ...` as well?

Reply:
Yes